### PR TITLE
MEN-5357: set default permissions to 0600 on device-config.json

### PR DESCRIPTION
**Pull request by tranchitella**
CHangelog: title

Signed-off-by: Fabio Tranchitella <fabio.tranchitella@northern.tech>

### CODE REVIEW

File: src/mender-configure [hunk 1]
```diff
@@ -37,7 +38,9 @@ safe_copy() {
         echo "safe_copy can only handle one file copy at a time" 1>&2
         exit 2
     fi
-    cp "$1" "$2".tmp
+    touch "$2".tmp
+    chmod $DEFAULT_PERMISSIONS "$2".tmp
+    cat "$1" > "$2".tmp
```
Comment (lluiscampos) [inline src/mender-configure:43]: What is the difference from this to:
```
cp "$1" "$2".tmp
chmod $DEFAULT_PERMISSIONS "$2".tmp
```
?

**Review by lluiscampos — APPROVED** [review]
Could you separate the `LICENSE` + `LIC_FILES_CHKSUM.sha256` changes to a separate commit? The source change is something that we might cherry-pick so it is cleaner if the commit contains only the actual fix.

Looks good otherwise.

File: src/mender-configure (hunk 1 above)
Comment (tranchitella) [reply src/mender-configure:43]: I didn't want to create the file in the target folder with the content we want to protect, readable by everybody, not even for a single moment. That's why I'm creating the empty file, changing the permissions, then copying the content over. Does it make sense?

**Comment by tranchitella** [comment]
> Could you separate the `LICENSE` + `LIC_FILES_CHKSUM.sha256` changes to a separate commit? The source change is something that we might cherry-pick so it is cleaner if the commit contains only the actual fix.

Done.

File: src/mender-configure (hunk 1 above)
Comment (lluiscampos) [reply src/mender-configure:43]: It makes sense, thanks for the clarification.

**Comment by tranchitella** [comment]
@mender-test-bot start pipeline

**Review by kacf — APPROVED** [review]

**Pull request merged**